packages/@aws-cdk/aws-ecs/test/cluster.test.ts

```diff
@@ -2155,6 +2155,98 @@ describe('cluster', () => {
 
   });
 
+  test('should throw an error if capacity provider with default strategy is not present in capacity providers', () => {
+    // GIVEN
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'test');
+
+    // THEN
+    expect(() => {
+      new ecs.Cluster(stack, 'EcsCluster', {
+        enableFargateCapacityProviders: true,
+      }).addDefaultCapacityProviderStrategy([
+        { capacityProvider: 'test capacityProvider', base: 10, weight: 50 },
+      ]);
+    }).toThrow(/Capacity provider with default strategy test capacityProvider is not present in cluster's capacity providers./);
+  });
+
+  test('should throw an error when capacity providers is length 0 and default capacity provider startegy specified', () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'test');
+
+    // THEN
+    expect(() => {
+      new ecs.Cluster(stack, 'EcsCluster', {
+        enableFargateCapacityProviders: false,
+      }).addDefaultCapacityProviderStrategy([
+        { capacityProvider: 'test capacityProvider', base: 10, weight: 50 },
+      ]);
+    }).toThrow(/Capacity provider with default strategy test capacityProvider is not present in cluster's capacity providers./);
+  });
+
+  test('should throw an error when more than 1 default capacity provider have base specified', () => {
+    const app = new cdk.App();
+    const stack = new cdk.Stack(app, 'test');
+
+    // THEN
+    expect(() => {
+      new ecs.Cluster(stack, 'EcsCluster', {
+        enableFargateCapacityProviders: true,
+      }).addDefaultCapacityProviderStrategy([
+        { capacityProvider: 'FARGATE', base: 10, weight: 50 },
+        { capacityProvider: 'FARGATE_SPOT', base: 10, weight: 50 },
+      ]);
+    }).toThrow(/Only 1 capacity provider with default strategy can have a nonzero base./);
+  });
+
+  test('can add ASG capacity via Capacity Provider with default capacity provider', () => {
```

Comment:
Please add another test case for success in the case where you 
1. add an ASG capacity provider to the cluster, then
2. add a defaultCapacityProviderStrategy 
3. which includes the ASG capacity provider
4. with the `addDefaultCapacityProviderStrategy` method

I'm not confident that your method works for ASG capacity providers since you aren't testing for it.